master/internal/checkpoint_gc.go

```diff
@@ -77,17 +102,31 @@ func (t *checkpointGCTask) Receive(ctx *actor.Context) error {
 	case actor.ChildFailed:
 		if msg.Child.Address().Local() == t.allocationID.String() {
 			t.completeTask(ctx)
+			return errors.Wrapf(msg.Error, "checkpoint GC task failed (actor.ChildFailed)")
 		}
 	case actor.PostStop:
 	default:
 		return actor.ErrUnexpectedMessage(ctx)
 	}
+
+	conv := &protoconverter.ProtoConverter{}
```

Comment:
this will run on the first message that gets through the switch case (so `actor.PreStart`) which isn't right. we want to run this only once the allocation actor we kick off to run the GC job has exited successfully, which we will know when we receive the `task.AllocationExited` and it has a nil error